### PR TITLE
Migrate IntImm & FloatImm ObjectRef to not-null

### PR DESCRIPTION
**Pull request by ANSHUMAN87**
Refer #5318

@tqchen, @jroesch, @zhiics, @junrushao1994 : Please help review, Thanks!

### CODE REVIEW

File: src/arith/rewrite_simplify.cc [hunk 1]
```diff
@@ -123,7 +123,7 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const AddNode* op) {
   // Pattern var to match any expression
   PVar<PrimExpr> x, y, z, b1, b2, s1, s2;
   // Pattern var match IntImm
-  PVar<IntImm> c1, c2, c3;
+  PVarOpt<Optional<IntImm>> c1, c2, c3;
```
Comment (tqchen) [inline src/arith/rewrite_simplify.cc:126]: I don't think it is right to use optional here, as the intention is to match the variable itself

Comment (ANSHUMAN87) [reply src/arith/rewrite_simplify.cc:126]: Yes, I agree. It is only to accommodate absent of default constructor. However the behaviour don't violate with PVar I think.

Comment (junrushao1994) [reply src/arith/rewrite_simplify.cc:126]: I agree too...Is there any better to do this?

Comment (ANSHUMAN87) [reply src/arith/rewrite_simplify.cc:126]: I have 2 possible approach here:

A0: Rename PVarOpt --> PVarExt(Extended) and hide Optional<T> input format.  
Look like --> `PVarExt<IntImm> c1, c2, c3;`

A1: Provide default constructor with default values. 
Look like: `PVar<IntImm> c1(IntImm(DataType::Int(32), 0)), c2(IntImm(DataType::Int(32), 0)), c3(IntImm(DataType::Int(32), 0));`

Please let me know your thoughts on this. Thanks!

Comment (tqchen) [reply src/arith/rewrite_simplify.cc:126]: I don;t think either approaches are good here. Let us leave it as it is for now

**Pull request closed without merging**